### ISSUE TITLE
Does the DKL really work?

### ISSUE THREAD

**Issue by kingaza** [cornellius-gp/gpytorch#301]
I try the example [combining KISS-GP, Deep Kernel Learning, and stochastic variational inference](https://github.com/cornellius-gp/gpytorch/blob/master/examples/05_Scalable_GP_Regression_Multidimensional/KISSGP_Deep_Kernel_Regression_With_Stochastic_Variational_Inference_CUDA.ipynb). The RMSE is 0.48. 
However, the y values are in [-1.25, 1.90]. Therefore, I think this RMSE is too large. This situation exists even if I increase the epoch number to 100.
Actually the predicted values are totally different with ground true values. 
I feel very confused...

**Comment by jacobrgardner** [comment]
In that notebook we are reporting mean absolute error, and the numbers with sufficient training are at least consistent for that dataset with the numbers reported in the DKL paper https://arxiv.org/pdf/1511.02222.pdf for the song dataset. That paper also reports the error of the neural network on its own, which is comparable. Therefore, we reasonably believe that DKL is indeed working as expected, even absent things like weight decay which @eytan mentioned.

The two main differences that lead to _very slightly_ higher MAE are (1) we actually use the smaller neural network `[d-1000-500-50-2]` instead of the larger one `[d-1000-1000-500-50-2]` just for the example notebook, and (2) the notebook only trains for a single epoch. It is easy to match the error by changing these things.

**Comment by jacobrgardner** [comment]
@eytan @gpleiss @andrewgordonwilson Our original design goal for the DKL example notebooks was just to reproduce closely the results from the original paper. Maybe it would be better for us to actually do some hyperparameter tuning and architecture design and just try to get results that are as competitive as possible for the benchmark datasets those notebooks download.

**Comment by kingaza** [comment]
thanks @eytan @jacobrgardner for the quick reply and suggestion
@jacobrgardner actually I have done 1) larger network and 2) more epochs, but the error is still too large. I will double check it again and tried decay and batchnorm.

thx again.

**Comment by eytan** [comment]
I wonder if it's worth having two separate notebooks, one that is a sort of collection of replications from the literature to show the sanity of the software, and another that demonstrates the state-of-the-art / best practices.  I suspect that many users (myself included) will start using the software by copypasta'ing bits of these tutorial notebooks, so it's important to add things like weight decay or additional hyperparameter tuning if we think that's important.

FWIW, I am currently trying to do some more systematic parameter sweep to better understand the effects of these different HPs.  Currently I am just doing it in the context of an internal application, but I can try to write it in a reasonably generic way so that we can evaluate this more generally with UCI datasets.
